src/v2/layouts/MemberDataGrid/MembersActionWrapper/CSVExportButton/CSVExportButton.jsx

```diff
@@ -0,0 +1,93 @@
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import moment from 'moment';
+import styled from 'styled-components';
+import { CSVLink } from 'react-csv';
+import { ExportIcon } from './ExportIcon';
+import { useContext } from 'react';
+import { StringsContext } from '../../../../components/providers/StringsProvider/StringsContext';
+
+const DisabledButtonStyled = styled.button`
+	margin: 0;
+	padding: 5px 10px;
+	display: ${(props) => (props.isView ? 'flex' : 'none')};
+	justify-content: center;
+	align-items: center;
+
+	svg {
+		width: 18px;
+		height: 18px;
+	}
+`;
+
+const CSVLinkStyled = styled(CSVLink)`
+	display: ${(props) => (props.disable === 'true' ? 'none' : 'flex')};
+	cursor: ${(props) => (props.disable === 'true' ? 'not-allowed' : null)};
+	justify-content: center;
+	align-items: center;
+	font-size: 14px;
+	border-radius: 4px;
+
+	&:active {
+		box-shadow: none;
+	}
+
+	svg {
+		width: 18px;
+		height: 18px;
+	}
+`;
+
+export const CSVExportButton = ({ metaData, clearSelectedRows, selectedMemberIds }) => {
+	const { strings } = useContext(StringsContext);
+	const { EXPORT } = strings;
+	const filteredHeader = metaData.headerConfig.filter((obj) => obj.excludeFromExport !== true);
+
+	const updatedSelectedMembers = [];
+	selectedMemberIds?.forEach((member) => {
+		filteredHeader.forEach((headerObj) => {
+			if (!member[headerObj.key]) {
+				member = {
+					...member,
+					[headerObj.key]: 'NA',
+				};
+			}
+		});
+		updatedSelectedMembers.push(member);
+	});
+
+	if (isEmpty(selectedMemberIds)) {
+		return (
+			<DisabledButtonStyled
+				isView={isEmpty(selectedMemberIds)}
```

Comment:
Also, this will only be rendered when we don't have selectedMemberIds. So I don't think there is a point of having softDisable and isView prop
```suggestion
				isView={!selectedMemberIds?.length}
```